### PR TITLE
Fixed travis maven deployment

### PR DESCRIPTION
**Pull request by fboucquez**
Hi Steve, can you review this pull request?

I missed a Gradle patch when I did the merge and I needed to move to create a release script to keep Travis happy.

I have also change Travis to only deploy the oracle jdk 8 artifacts

### CODE REVIEW

**Comment by fboucquez** [comment]
Github hides env variables when doing a fork's PR due to security concern. It makes sense, a hacky PR that comes from a fork could just sys out a secured env credential/variable. I removed an invalid validation to fix when there is no env variable set for deployments (PRs doesn't do deployments, just master once merged).

When enabling sonar, we may need to only do PRs from this repository (not forks). If not, PR won't be able to publish the sonar reports that will allow us accepting or rejecting the PR (the PR won't see the sonar security credentials from the env variables)

This PR is fine, @rg911 have a look when you have the time.

File: release.sh [hunk 1]
```diff
@@ -0,0 +1,8 @@
+#!/usr/bin/env bash
+set -e
+
+echo "Uploading generated clients jars"
+./gradlew -b ./openapi-generator/build.gradle uploadArchives
+
+echo "Uploading sdk jars"
+./gradlew uploadArchives
```
Comment (rg911) [inline release.sh]: trailing line after this?

File: openapi-generator/build.gradle [hunk 2]
```diff
@@ -59,40 +77,39 @@ subprojects {
...
             mavenDeployer {
-                if (project.hasProperty("signing.keyId")) {
+
+                if (signingKeyId != null) {
                     beforeDeployment { MavenDeployment deployment -> signing.signPom(deployment) }
                 }
 
                 repository(url: "https://oss.sonatype.org/service/local/staging/deploy/maven2/") {
-                    authentication(userName: username, password: password)
+                    authentication(userName: ossrhUsername, password: ossrhPassword)
```
Comment (rg911) [inline openapi-generator/build.gradle]: where are these ossrhUsernames and password get saved?

Comment (fboucquez) [reply openapi-generator/build.gradle]: Travis env variables when running from Travis.

<user_home>/.gradle/gradle.properties when running locally.

File: release.sh (hunk 1 above)
Comment (fboucquez) [reply release.sh]: New line added

**Pull request merged**